src/intertypes/json.jl

```diff
@@ -242,68 +263,75 @@ function write(io::IO, format::JSONFormat, acs::ACSet)
   end
 end
 
-const Object = OrderedDict{String, Any}
-
 function fieldproperties(fields::Vector{Field{InterType}})
   map(fields) do field
-    string(field.name) => tojsonschema(field.type)
+    field.name => tojsonschema(field.type)
   end
 end
 
 function tojsonschema(type::InterType)
   @match type begin
     I32 => Object(
-      "type" => "integer",
-      "\$comment" => "I32",
-      "minimum" => typemin(Int32),
-      "maximum" => typemax(Int32)
+      :type => "integer",
+      Symbol("\$comment") => "I32",
+      :minimum => typemin(Int32),
+      :maximum => typemax(Int32)
     )
     U32 => Object(
-      "type" => "integer",
-      "\$comment" => "U32",
-      "minimum" => typemin(UInt32),
-      "maximum" => typemax(UInt32)
+      :type => "integer",
+      Symbol("\$comment") => "U32",
+      :minimum => typemin(UInt32),
+      :maximum => typemax(UInt32)
     )
     I64 => Object(
-      "type" => "string",
-      "\$comment" => "I64"
+      :type => "string",
+      Symbol("\$comment") => "I64"
     )
     U64 => Object(
-      "type" => "string",
-      "\$comment" => "U64"
+      :type => "string",
+      Symbol("\$comment") => "U64"
     )
     F64 => Object(
-      "type" => "number",
-      "\$comment" => "F64"
+      :type => "number",
+      Symbol("\$comment") => "F64"
     )
     Boolean => Object(
-      "type" => "boolean",
-      "\$comment" => "Boolean"
+      :type => "boolean",
+      Symbol("\$comment") => "Boolean"
     )
     Str => Object(
-      "type" => "string",
-      "\$comment" => "Str"
+      :type => "string",
+      Symbol("\$comment") => "Str"
     )
     Sym => Object(
-      "type" => "string",
-      "\$comment" => "Sym"
+      :type => "string",
+      Symbol("\$comment") => "Sym"
     )
     Binary => Object(
-      "type" => "string",
-      "contentEncoding" => "base64",
-      "\$comment" => "Binary"
+      :type => "string",
+      :contentEncoding => "base64",
+      Symbol("\$comment") => "Binary"
+    )
+    OptionalType(elemtype) => begin
+      schema = tojsonschema(elemtype)
+      schema[:type] = [schema[:type], "null"]
```

Comment:
The schema of a type can be a list including the string null? Is this a JSONSchema idiom?

Reply:
I think so? This is what showed up on the jsonschema website when I poked around for it, but I'm no jsonschema expert.

Reply:
@fivegrant can probably confirm that it is what he would expect for a nullable field.